# general optimization params
train_batch_size: 64
val_batch_size: 16

# logging params
num_epochs: 50        # number of epochs to train for
eval_frequency: 10    # how often to get validation error (in epochs)
save_frequency: 10    # how often to save output (in epochs)
vis_frequency: 10000  # how often to visualize filters (in epochs)
log_frequency: 1      # how often to log output (in steps)

# train / val split params 
train_pct: 0.8              # percentage of the data to use for training vs validation
total_pct: 1.0              # percentage of all the files to use
eval_total_train_error: 0   # whether or not to evaluate the total training error on each validation
max_files_eval: 1000        # the number of validation files to use in each eval

# optimization params
loss: sparse
optimizer: momentum
train_l2_regularizer: 0.0005
base_lr: 0.01
decay_step_multiplier: 0.6   # proportion of training datapoints to see before stepping down learning rate (in epochs)
decay_rate: 0.95
momentum_rate: 0.9
max_training_examples_per_load: 128
drop_rate: 0.0
max_global_grad_norm: 100000000000

# fine-tuning params
fine_tune: 0
update_fc_only: 0
update_conv0_only: 0
reinit_pc1: 0
reinit_fc3: 0
reinit_fc4: 0
reinit_fc5: 0

# input params
training_mode: classification
image_field_name: tf_depth_ims
pose_field_name: grasps

# label params
target_metric_name: grasp_metrics  # name of the field to use for the labels
metric_thresh: 0.75                 # threshold for positive examples (label = 1 if grasp_metric > metric_thresh)

# preproc params
num_random_files: 1000     # the number of random files to compute dataset statistics in preprocessing (lower speeds initialization)
preproc_log_frequency: 100 # how often to log preprocessing (in steps)

# denoising / synthetic data params
multiplicative_denoising: 0
gamma_shape: 1000.00

symmetrize: 1

gaussian_process_denoising: 0
gaussian_process_rate: 0.5
gaussian_process_scaling_factor: 4.0
gaussian_process_sigma: 0.005

# tensorboard
tensorboard_port: 6006

# debugging params
debug: 0
debug_num_files: 1000000000
seed: 24098

### GQCNN CONFIG ###
gqcnn:
  # basic data metrics
  im_height: 96
  im_width: 96
  im_channels: 1

  # needs to match input data mode that was used for training, determines the pose dimensions for the network
  gripper_mode: parallel_jaw

  # prediction batch size, in training this will be overriden by the val_batch_size in the SGDOptimizer's config file
  batch_size: 16

  # debugging params
  debug: 0
  seed: 24098

  # architecture
  architecture:
    im_stream:
      conv1_1:
        type: conv
        filt_dim: 9
        num_filt: 16
        pool_size: 2
        pool_stride: 2
        norm: 0
        pad: VALID
      conv1_2:
        type: conv
        filt_dim: 3
        num_filt: 16
        pool_size: 2
        pool_stride: 2
        norm: 1
        pad: VALID
      conv2_1:
        type: conv
        filt_dim: 3
        num_filt: 16
        pool_size: 1
        pool_stride: 1
        norm: 0
        pad: VALID
      conv2_2:
        type: conv
        filt_dim: 3
        num_filt: 16
        pool_size: 1
        pool_stride: 1
        norm: 1
        pad: VALID
      fc3:
        type: fc
        out_size: 32
    pose_stream:
      pc1:
        type: pc
        out_size: 4
    merge_stream:
      fc4:
        type: fc_merge
        out_size: 32
      fc5:
        type: fc
        out_size: 2

  # architecture normalization constants
  radius: 2
  alpha: 2.0e-05
  beta: 0.75
  bias: 1.0

  # leaky relu coefficient
  relu_coeff: 0.0
